lib/zstd.h

```diff
@@ -976,6 +978,22 @@ typedef enum {
     /* advanced parameters - may not remain available after API update */
     ZSTD_p_forceMaxWindow=1100, /* Force back-reference distances to remain < windowSize,
                               * even when referencing into Dictionary content (default:0) */
+    ZSTD_p_longDistanceMatching,  /* Enable long distance matching. This
```

Comment:
So, it's a `0 / 1` switch ?
Maybe `enableLongDistanceMatching` in this case ?

Also : try to describe what happens if expected setting order is not respected,
i.e. if `compressionLevel` is set after `enableLongDistanceMatching`, what happens ?

Reply:
I'm currently going with the "long distance matching has to be toggled on/off explicitly" semantics, though let me know if you think it makes more sense to be reset with `compressionLevel`. I'll add comments.

Reply:
I think that's fine.
As long as the logic is consistent, and clearly described, it's a good choice.

Reply:
Should we set `ZSTD_p_longDistanceMatching=1200`?

Reply:
The enum will need a pinned value before reaching "stable" status,
but at this stage, this is still pretty experimental, so it's not too important.
Better expect to change it sometimes in the future.